// Common types for Terminal gRPC service.
//
// Version: 1.0.0
// Date: 2025-12-04

syntax = "proto3";

package terminal;

option go_package = "terminal/proto";

import "google/protobuf/timestamp.proto";

// Terminal session status
enum SessionStatus {
  SESSION_STATUS_UNSPECIFIED = 0;
  PENDING = 1;
  CONNECTED = 2;
  DISCONNECTED = 3;
  ERROR = 4;
}

// Command execution status
enum CommandStatus {
  COMMAND_STATUS_UNSPECIFIED = 0;
  CMD_PENDING = 1;
  CMD_RUNNING = 2;
  CMD_SUCCESS = 3;
  CMD_FAILED = 4;
  CMD_CANCELLED = 5;
  CMD_TIMEOUT = 6;
}

// Log levels
enum LogLevel {
  LOG_LEVEL_UNSPECIFIED = 0;
  LOG_DEBUG = 1;
  LOG_INFO = 2;
  LOG_WARNING = 3;
  LOG_ERROR = 4;
}

// Terminal dimensions
message TerminalSize {
  int32 rows = 1;      // Number of rows
  int32 cols = 2;      // Number of columns
  int32 width = 3;     // Width in pixels (optional)
  int32 height = 4;    // Height in pixels (optional)
}

// Session configuration
message SessionConfig {
  string session_id = 1;           // UUID of the session
  string shell = 2;                // Shell to use (e.g., /bin/zsh)
  string working_directory = 3;    // Initial working directory
  map<string, string> env = 4;     // Environment variables
  TerminalSize size = 5;           // Terminal dimensions
}
